test/suite/hook_tiered.py

```diff
@@ -300,89 +300,36 @@ def uses(self, use_list):
-                "test_bug018.test_bug018",
-                "test_checkpoint.test_checkpoint",
-                "test_checkpoint_snapshot02.test_checkpoint_snapshot_with_txnid_and_timestamp",
-                "test_compat05.test_compat05",
-                "test_config05.test_too_many_sessions",
-                "test_config09.test_config09",
-                "test_drop.test_drop",
-                "test_empty.test_empty",     # looks at wt file names and uses column store
-                "test_encrypt06.test_encrypt",
-                "test_encrypt07.test_salvage_api",
-                "test_encrypt07.test_salvage_api_damaged",
-                "test_encrypt07.test_salvage_process_damaged",
-                "test_export01.test_export_restart",
-                "test_hs21.test_hs",
-                "test_import04.test_table_import",
-                "test_import09.test_import_table_repair",
-                "test_import09.test_import_table_repair",
-                "test_import11.test_file_import",
-                "test_import11.test_file_import",
-                "test_join03.test_join",
-                "test_join07.test_join_string",
-                "test_jsondump02.test_json_all_bytes",
-                "test_metadata_cursor02.test_missing",
-                "test_prepare02.test_prepare_session_operations",
-                "test_prepare_hs03.test_prepare_hs",
-                "test_prepare_hs03.test_prepare_hs",
-                "test_rename.test_rename",
-                "test_rollback_to_stable09.test_rollback_to_stable",
-                "test_rollback_to_stable28.test_update_restore_evict_recovery",
-                "test_rollback_to_stable34.test_rollback_to_stable",
-                "test_rollback_to_stable35.test_rollback_to_stable",
-                "test_rollback_to_stable36.test_rollback_to_stable",
-                "test_sweep03.test_disable_idle_timeout_drop",
-                "test_sweep03.test_disable_idle_timeout_drop_force",
-                "test_txn22.test_corrupt_meta",
-                "test_verbose01.test_verbose_single",
-                "test_verbose02.test_verbose_single",
-                "test_verify2.test_verify_ckpt",
-                ]
-
-        for item in skip:
-            if item in str(test):
-                return True
-        return False
-
-    # Remove tests that won't work on tiered cursors
-    def filter_tests(self, tests):
-        new_tests = unittest.TestSuite()
-        new_tests.addTests([t for t in tests if not self.skip_test(t)])
-        return new_tests
+    # Determine if a test should be skipped. 
+    # If it should skip, return the reason as a string. If it shouldn't, return None
+    # Some features aren't supported with tiered storage currently. If they exist in 
+    # the test name (or scenario name) skip the test.
+    def skip_reason(self, test):
+        skip_categories = [
+            ("backup",               "Can't backup a tiered table"),
```

Comment:
I wonder if the reasons should have a prefix such as `Tiered` in this case.
The first reason states `tiered table`, then the second one mentions `tiered storage`, etc. It deals with the same issue but we refer to the problem differently.

I could suggest:
```
"backup", "Tiered: Backup is not supported"
"inmem", "Tiered: In-memory tests are incompatble",
"lsn", "Tiered: Ignore LSM tests"
etc
```

What do you think?

Reply:
Here's an example output for a skipped test with tiered: 
`[pid:3614589]: Test salvage in a 'wt' process, using an empty table (column-fix.no-fail-point) ... skipped "test_salvage01.test_salvage01.test_salvage_process_empty -s 5 (column-fix.no-fail-point) for hook tiered: Salvage tests directly name files ending in '.wt'"`
where we get the tiered tagging for free.

I don't mind cleaning up the skip reasons though. Some small changes made in b1b6edcbbd64a24488d5e16f32797a4fee3b87af

Reply:
I did not know we would see the `for hook tiered`, that's good enough, thank you!